### PR TITLE
717 - Update config files for k8s

### PR DESCRIPTION
**Pull request by alaudazzi**
Needed for #717 
Blocked by #32248

Description
-----------

1. Added the k8s alias to the docs/doc_build_aliases.sh
2. Added the K8s book to the yaml. file

Documentation
-------------

- [x] User-facing Documentation
- [ ] Internal Documentation

@pebrc

### CODE REVIEW

File: conf.yaml [hunk 1]
```diff
@@ -523,6 +523,25 @@ contents:
                 path:   docs/shared
                 exclude_branches: [ 1.0 ]
 
+         -
+            title:      Elastic Cloud on Kubernetes (ECK)
```
Comment (nrichers) [inline conf.yaml]: Do we want to include `(ECK)` in the title here? I would imagine we probably want to just use this in the docs themselves.

File: doc_build_aliases.sh [hunk 2]
```diff
@@ -67,6 +67,8 @@ alias docbldec='$GIT_HOME/docs/build_docs.pl --doc $GIT_HOME/cloud/docs/saas/ind
 
 alias docbldece='$GIT_HOME/docs/build_docs.pl --doc $GIT_HOME/cloud/docs/cloud-enterprise/index.asciidoc --resource=$GIT_HOME/cloud/docs/shared --chunk 1'
 
+alias docbldk8s='$GIT_HOME/docs/build_docs.pl --doc $GIT_HOME/cloud/docs/k8s/index.asciidoc --chunk 1'
```
[the remark this answers is not in the document]
Comment (nik9000) [reply doc_build_aliases.sh]: I prefer `--resource $GIT_HOME/cloud/docs/shared` without the `=`. Not a big deal. I just like it better.

**Comment by nik9000** [comment]
You should be able to test the `conf.yaml` change locally with:
```
./build_docs --all --target_repo git@github.com:elastic/build-docs.git --open
```

It will take a long, long time. First it has to `git fetch` about 4GB worth of repos, then it rebuilds all of the books that have changed since the last time they were built by CI. That is usually a few books. Eventually it'll get to your book.... I'm sorry this process is so slow!

**Comment by nik9000** [comment]
Oh! You'll probably want to use your local checkout of cloud:
```
./build_docs --all --target_repo git@github.com:elastic/build-docs.git --open --sub_dir cloud:master:../cloud
```

File: conf.yaml [hunk 3]
```diff
@@ -523,6 +523,25 @@ contents:
...
+            prefix:     en/k8s
+            tags:       Kubernetes/Reference
+            subject:    k8s
+            current:    2.2
+            branches:   [ ]
+            index:      docs/k8s/index.asciidoc
```
Comment (nrichers) [inline conf.yaml]: Note that this PR intends to build the quick start out of the K8s repo, but https://github.com/elastic/cloud/pull/32248 targets the Cloud repo. (https://github.com/elastic/cloud/pull/32248 needs to be replaced by a new PR that targets the K8s repo.)

File: conf.yaml (hunk 1 above)
Comment (alaudazzi) [reply conf.yaml]: Done.

File: conf.yaml [hunk 4]
```diff
@@ -523,6 +523,25 @@ contents:
                 path:   docs/shared
                 exclude_branches: [ 1.0 ]
 
+         -
+            title:      Elastic Cloud on Kubernetes (ECK)
+            prefix:     en/k8s
+            tags:       Kubernetes/Reference
+            subject:    k8s
+            current:    2.2
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply conf.yaml]: Done.

File: conf.yaml [hunk 5]
```diff
@@ -523,6 +523,25 @@ contents:
...
+            branches:   [ ]
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply conf.yaml]: Done.

File: conf.yaml [hunk 6]
```diff
@@ -523,6 +523,25 @@ contents:
...
+            chunk:      1
+            private:    1
+            sources:
+              -
+                repo:   cloud
+                path:   docs/k8s
+              -
+                repo:   cloud
+                path:   docs/shared
+                exclude_branches: [ 1.0 ]
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply conf.yaml]: OK. I leave it as is.

File: doc_build_aliases.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (alaudazzi) [reply doc_build_aliases.sh]: Done.

[the remark this answers is not in the document]
Comment (alaudazzi) [reply doc_build_aliases.sh]: I also updated the path according to the latest k8s repo renaming, which is:
$GIT_HOME/cloud-on-k8s/docs/index.asciidoc

File: conf.yaml [hunk 7]
```diff
@@ -525,6 +525,21 @@ contents:
...
+         -
+            title:      Elastic Cloud on Kubernetes
+            prefix:     en/k8s
+            tags:       Kubernetes/Reference
+            subject:    k8s
+            current:    master
+            branches:   [ master ]
+            index:      docs/index.asciidoc
+            chunk:      1
+            public:     1
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply conf.yaml]: Where exactly do I need to add `asciidoctor: true`?

File: doc_build_aliases.sh [hunk 8]
```diff
@@ -67,6 +67,8 @@ alias docbldec='$GIT_HOME/docs/build_docs --doc $GIT_HOME/cloud/docs/saas/index.
 
 alias docbldece='$GIT_HOME/docs/build_docs --doc $GIT_HOME/cloud/docs/cloud-enterprise/index.asciidoc --resource=$GIT_HOME/cloud/docs/shared --chunk 1'
 
+alias docbldk8s='$GIT_HOME/docs/build_docs --doc $GIT_HOME/cloud-on-k8s/docs/index.asciidoc --resource $GIT_HOME/cloud/docs/shared --chunk 1'
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply doc_build_aliases.sh]: Where do I need to add `--asciidoctor`?

**Comment by lcawl** [comment]
@alaudazzi and I chatted and the command in the aliases file works now.
I am also running the command to build all books ($GIT_HOME/docs/build_docs --all --target_repo git@github.com:elastic/built-docs.git --open) and it's in progress (so far so good).

File: conf.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (nik9000) [reply conf.yaml]: If you replace this line with `asciidoctor: true` then it'll build with asciidoctor. `public: 1` doesn't look like it does anything.

File: doc_build_aliases.sh (hunk 8 above)
[the remark this answers is not in the document]
Comment (nik9000) [reply doc_build_aliases.sh]: If you add it right after `build_docs` then it'll line up with the other lines in this file. But anywhere in the line is actually fine.

**Comment by lcawl** [comment]
FYI When I build ALL the docs, it returns the following broken links:
INFO:build_docs:Bad cross-document links:
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/k8s/current/k8s-deploy_the_elasticsearch_cluster.html:
INFO:build_docs:   - en/elasticsearch/reference/7.0getting-started.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/k8s/current/k8s-deploy_the_kibana_instance.html:
INFO:build_docs:   - en/kibana/7.0introduction.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/k8s/master/k8s-deploy_the_elasticsearch_cluster.html:
INFO:build_docs:   - en/elasticsearch/reference/7.0getting-started.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/k8s/master/k8s-deploy_the_kibana_instance.html:
INFO:build_docs:   - en/kibana/7.0introduction.html

**Comment by alaudazzi** [comment]
A big THANK YOU to all of you for the great support you provided!

File: conf.yaml [hunk 9]
```diff
@@ -525,6 +526,21 @@ contents:
                 path:   docs/shared
                 exclude_branches: [ 1.0 ]
 
+         -
+            title:      Elastic Cloud on Kubernetes
+            prefix:     en/k8s
```
Comment (pebrc) [inline conf.yaml]: Can we make this prefix match the repo name? `cloud-on-k8s`

Comment (alaudazzi) [reply conf.yaml]: Done.

**Pull request merged**